content/docs/guides/adopting/from_azure.md

```diff
@@ -0,0 +1,703 @@
+---
+title: "From Azure Resource Manager (ARM)"
+meta_desc: Migrate your existing Azure Resource Manager (ARM) templates and/or coexist with existing deployments.
+menu:
+  userguides:
+    parent: adopting
+    weight: 4
+---
+
+<img src="/logos/tech/azure_arm.png" align="right" class="h-32 px-8 pb-4">
+
+If your team has already provisioned infrastructure using Azure Resource Manager (ARM) Templates, and you'd like to adopt Pulumi, you have three primary strategies you can take:
+
+* [**Coexist**](#referencing-deployment-outputs) with resources provisioned by ARM by referencing deployment outputs.
+* [**Import**]({{< relref "import" >}}) existing resources into Pulumi in the usual way.
+* [**Convert**](#converting-deployments-and-resources) your deployments to use Pulumi and then incrementally migrate resources.
+
+## Referencing Stack Outputs
+
+It is possible to reference existing Azure Resource Manager (ARM) template deployments from your program. It doesn't mattter how these templates and deployments were created. This lets you read properties of a deployment for use within your Pulumi program. This includes output values computed from resources provisioned that stack.
+
+For instance, let's say your infrastructure team has provisioned your Azure storage account using ARM and you need to use the Storage Account name to provision something new from your Pulumi program. One approach is to hardcode the name but this is brittle and, if it ever changes, you'd need to go and manually update the hardcoded value.
+
+Instead, you can look up that ARM deployment by name and use one of its output values. The following example reads a deployment by its fully qualified ID and then uses the exported `storageAccountName` value to upload a private zipfile to blob storage, containing a `wwwroot/` directory locally:
+
+{{< langchoose csharp >}}
+
+```javascript
+let azure = require("@pulumi/azure");
+let pulumi = require("@pulumi/pulumi");
+
+// Read the deployment and the storage account name.
+let deployment = azure.core.TemplateDeployment.get("myStorageDeployment",
+    "/subscriptions/0292631f-7a9b-4142-90b2-96badd5eafa8/resourceGroups/myrg8fd69ec2/providers/Microsoft.Resources/deployments/myStorageDeployment62ba53a3");
+let storageAccountName = deployment.outputs["storageAccountName"];
+
+// Create a blob for our own deployment.
+let blob = new azure.storage.ZipBlob("myBlob", {
+    storageAccountName: storageAccountName,
+    storageContainerName: new azure.storage.Container("myStorageContainer", {
+        storageAccountName: storageAccountName,
+        containerAccessType: "private",
+    }).name,
+    type: "block",
+    content: new pulumi.asset.FileArchive("wwwroot"),
+});
```

Comment:
I think we're just about to remove `ZipBlob` as part of the 2.0 Azure provider (it was a hack to work around Archive/Asset issues).

Unfortunately, thinking about this example, I'm not sure there's code we can write that works both before and after those changes.  @mikhailshilkov is that right? 

https://github.com/pulumi/pulumi-azure/pull/479

Reply:
Ahh, shucks. I was trying to find a simple pair of resources that would go nicely together -- similar to the CloudFormation pair of VPC/Subnet and WebServer. Does anything else come to mind?

Reply:
I'll go ahead and merge as-is (especially as it took an hour or so to create and run all of these variants for different languages), but if you guys have a better replacement, please do let me know.